modules/storage-gcs/gcs/gcs.go

```diff
@@ -241,12 +269,36 @@ func (g *gcs) SetMetaStatus(ctx context.Context, className, snapshotID, status s
 
 func (g *gcs) DestinationPath(className, snapshotID string) string {
 	return "gs://" + path.Join(g.config.BucketName(),
-		makeObjectName(className, snapshotID, "snapshot.json"))
+		g.makeObjectName(className, snapshotID, "snapshot.json"))
 }
 
 func (g *gcs) InitSnapshot(ctx context.Context, className, snapshotID string) (*snapshots.Snapshot, error) {
-	// TODO: implement
-	return nil, nil
+	bucket, err := g.findBucket(ctx)
+	if err != nil && !errors.Is(err, snapshots.ErrNotFound{}) {
+		return nil, errors.Wrap(err, "init snapshot")
+	}
+
+	if bucket == nil {
+		bucket, err = g.createBucket(ctx)
```

Comment:
I suggest that the user create the bucket and the root folder himself to avoid access rights conflicts.
In consequence the user would grant access for weaviate to access the root folder.
In some companies this might be restricted to administrators only.

Reply:
Good point

Reply:
I believe this change would make `DEFAULT_BUCKET` obsolete. In this case, we would need to make  `STORAGE_GCS_BUCKET` a required variable. 

Thoughts @antas-marcin ?

Reply:
Spoke offline with @antas-marcin about this, we agreed to make it a requirement that a provided bucket must exist for this operation to be successful. I will make this change in a separate PR